lib/fs/fakefs.go

```diff
@@ -218,6 +227,9 @@ func (fs *fakefs) create(name string) (*fakeEntry, error) {
 		entry.size = 0
 		entry.mtime = time.Now()
 		entry.mode = 0666
+		if fs.insens {
+			entry.name = filepath.Base(name)
+		}
```

Comment:
You have an issue again, where the name leaks between two create calls. Call create with foo, store ref, call create with Foo, I expect name of the first ref to change.

Reply:
It does indeed change, see the updated test. Or do you expect it to remain `foo`?

Reply:
I do not expect filenames of things we already have descriptors for to change just because someone else opened the file with a different case.

Reply:
I think we really need to run those tests on a real fs on windows (or mac) to validate the behaviour of an actual case-insensitive fs: see https://github.com/syncthing/syncthing/pull/6074#issuecomment-545316396.

Reply:
> I do not expect filenames of things we already have descriptors for to change just because someone else opened the file with a different case.

In this case someone didn't just open the file, but actually successfully created a new file it that place. I agree, we need to test the real behaviour to have an idea what to really shoot for.